pkg/resolution/resolver/git/resolver_test.go

```diff
@@ -489,16 +509,19 @@ func TestResolve(t *testing.T) {
 						expectedStatus.Annotations = make(map[string]string)
 					}
 					expectedStatus.Annotations[resolutioncommon.AnnotationKeyContentType] = "application/x-yaml"
-					switch {
-					case tc.useNthCommit > 0:
-						expectedStatus.Annotations[AnnotationKeyRevision] = commits[plumbing.Master.Short()][tc.useNthCommit]
-					case tc.revision == "" && reqParams[urlParam] != "":
-						expectedStatus.Annotations[AnnotationKeyRevision] = plumbing.Master.Short()
-					case tc.revision == "":
-						expectedStatus.Annotations[AnnotationKeyRevision] = tc.config[defaultRevisionKey]
-					default:
-						expectedStatus.Annotations[AnnotationKeyRevision] = tc.revision
+
+					var expectedCommitSHAInStatus string
+					// Anonymous cloning
+					if len(tc.commits) > 0 {
+						expectedCommitSHAInStatus = commitSHAs[tc.expectedNthCommit]
+					} else if tc.revision != "" {
+						// scm cloning
+						expectedCommitSHAInStatus = tc.revision
```

Comment:
Not blocking if you want to push forward with this as-is, but wdyt about simplifying this further? I think something like this would make it easier to see what's going on:

```go
repo := createTestRepo(...)

// map revision -> resolved SHA
// SHAs should be deterministic if timestamps are static.
tc := map[string]string{
  "": "abcd",
  "abcd": "abc",
  "main": "abc",
  "other-branch": "def",
  "main^": "def",
  "refs/tags/v1": "def",
  "refs/other/asdf": "xyz",
}

for k, v := range want {
  req := createRequest(repo, k)
  
 resp := new(v1beta1.ResolutionRequestStatus)
 RunResolverReconcileTest(..., got, ...)  // or call resolver directly

 want  := &v1beta1.ResolutionRequestStatus{
    ...
    Annotations: map[string]string{
      AnnotationKeyRevision: v,
      ...
    }
  }

  if cmp.Diff(want, got) {
     ...
  }
}
```

Reply:
Thanks a lot for the suggestion here @wlynch !!! Really appreciate it!

> repo := createTestRepo(...)

Creating a local test repo in advance with some pre-defined commits rather than per-test commits is a good idea. Done!  (Note this setting up local test repo is for anonymous clone approach only. SCM local repo set up is a bit different.)

> // map revision -> resolved SHA. SHAs should be deterministic if timestamps are static.

SG! Done!

> RunResolverReconcileTest(..., got, ...)  // or call resolver directly

Perhaps we should keep using `RunResolverReconcileTest` because I found in a previous commit in which a unit test [calls `resolve` directly](https://github.com/tektoncd/pipeline/blob/6074fa43e2381201ba94576ad1cfe815312ae228/pkg/resolution/resolver/git/resolver_test.go#L268), but they all consolidated to `RunResolverReconcileTest` at some point. So it might be a good idea to keep using RunResolverReconcileTest, which is consistent with how other resolvers are tested.


With regards to the test table, I think we should have keep existing fields other than having the revision itself because we do need to test some failure cases i.e path not found, scm token etc. 

That said, I've made some changes to the test table to make it look a bit cleaner. Please take a look and let me know wdyt. ty!